### PR TITLE
Add ADC support for stm32h735g_disco

### PR DESCRIPTION
**Pull request by besmarsh**
Add ADC support for stm32h735g_disco in devicetree and to enable ADC sample to be built for this board. Please let me know if I've missed anything!

This PR also modifies the ADC sample to correctly display readings from 16-bit resolution ADC channels.

### CODE REVIEW

**Comment by erwango** [comment]
@besmarsh Please rebase

File: boards/arm/stm32h735g_disco/stm32h735g_disco.dts [hunk 1]
```diff
@@ -98,6 +98,12 @@
 	pinctrl-names = "default";
 };
 
+&adc1 {
+	pinctrl-0 = < &adc1_inp0_pa0_c>;
```
Comment (erwango) [inline boards/arm/stm32h735g_disco/stm32h735g_disco.dts:102]: Nit
```suggestion
	pinctrl-0 = <&adc1_inp0_pa0_c>;
```

**Review by erwango — CHANGES_REQUESTED** [review]

File: samples/drivers/adc/boards/stm32h735g_disco.overlay [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/* Copyright (c) 2021 STMicroelectronics
```
Comment (FRASTM) [inline samples/drivers/adc/boards/stm32h735g_disco.overlay:1]: Copyright ?

**Review by FRASTM — COMMENTED** [review]
LGTM but rebase and CI to fix

File: samples/drivers/adc/boards/stm32h735g_disco.overlay (hunk 2 above)
Comment (besmarsh) [reply samples/drivers/adc/boards/stm32h735g_disco.overlay:1]: I copied the notice from another board's overlay, I'm not familiar with the process for copyright notices. What should I change here?

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Comment by besmarsh** [comment]
Thanks for the reviews and merging! Does this need adding to the v3.4 milestone?

**Pull request merged**